lib/cocoapods/installer/analyzer.rb

```diff
@@ -682,6 +690,13 @@ def dependencies_for_specs(specs, platform, all_specs)
         specs.each do |s|
           s.dependencies(platform).each do |dep|
             all_specs[dep.name].each do |spec|
+              if spec.non_library_specification?
+                if s.test_specification? && spec.name == s.consumer(platform).app_host_name && spec.app_specification?
```

Comment:
nit of the nits but we've seen performance problems in the past, is it worthwhile to check for `spec.app_specification?` before we do `==` check? Probably micro optimizing but I've been bitten before in the past.

Reply:
probably won't make a difference, since by that point in the conditional we already have a test spec depending on another non-library spec.